samza-core/src/main/java/org/apache/samza/execution/JobNodeConfigurationGenerator.java

```diff
@@ -190,68 +198,27 @@ private long computeTriggerInterval(Collection<OperatorSpec> reachableOperators)
-    // Disallow user specified job inputs/outputs. This info comes strictly from the user application.
-    Map<String, String> allowedConfigs = new HashMap<>(config);
-    if (!jobNode.isLegacyTaskApplication()) {
-      if (allowedConfigs.containsKey(TaskConfig.INPUT_STREAMS())) {
-        LOG.warn("Specifying task inputs in configuration is not allowed for SamzaApplication. "
-            + "Ignoring configured value for " + TaskConfig.INPUT_STREAMS());
-        allowedConfigs.remove(TaskConfig.INPUT_STREAMS());
-      }
-    }
-
-    LOG.debug("Job {} has allowed configs {}", jobNode.getJobNameAndId(), allowedConfigs);
-    return mergeJobConfig(new MapConfig(allowedConfigs), new MapConfig(configs));
-  }
-
-  /**
-   * This function extract the subset of configs from the full config, and use it to override the generated configs
-   * from the job.
-   * @param fullConfig full config
-   * @param generatedConfig config generated for the job
-   * @param configPrefix prefix to extract the subset of the config overrides
-   * @return config that merges the generated configs and overrides
-   */
-  private static Config extractScopedConfig(Config fullConfig, Config generatedConfig, String configPrefix) {
-    Config scopedConfig = fullConfig.subset(configPrefix);
-
-    Config[] configPrecedence = new Config[] {fullConfig, generatedConfig, scopedConfig};
-    // Strip empty configs so they don't override the configs before them.
-    Map<String, String> mergedConfig = new HashMap<>();
-    for (Map<String, String> config : configPrecedence) {
-      for (Map.Entry<String, String> property : config.entrySet()) {
-        String value = property.getValue();
-        if (!(value == null || value.isEmpty())) {
-          mergedConfig.put(property.getKey(), property.getValue());
-        }
-      }
-    }
-    scopedConfig = new MapConfig(mergedConfig);
-    LOG.debug("Prefix '{}' has merged config {}", configPrefix, scopedConfig);
-
-    return scopedConfig;
-  }
-
   private List<StoreDescriptor> getStoreDescriptors(Collection<OperatorSpec> reachableOperators) {
     return reachableOperators.stream().filter(operatorSpec -> operatorSpec instanceof StatefulOperatorSpec)
         .map(operatorSpec -> ((StatefulOperatorSpec) operatorSpec).getStoreDescriptors()).flatMap(Collection::stream)
         .collect(Collectors.toList());
   }
 
-  private void configureTables(Map<String, String> configs, Config config, Map<String, TableSpec> tables, Set<String> inputs) {
-    configs.putAll(TableConfigGenerator.generateConfigsForTableSpecs(new MapConfig(configs),
-        tables.values().stream().collect(Collectors.toList())));
+  private void configureTables(Map<String, String> generatedConfig, Config originalConfig,
+      Map<String, TableSpec> tables, Set<String> inputs) {
+    generatedConfig.putAll(
+        TableConfigGenerator.generateConfigsForTableSpecs(
+            new MapConfig(generatedConfig), new ArrayList<>(tables.values())));
```

Comment:
Instead of passing generatedConfig, should this pass "config" (or even possibly config merged with the current state of generatedConfig)? I didn't look too closely into it, but something looks like it might be wrong based on the initial read.

Reply:
Ah yes, your changes aren't changing any functionality. However, now that you've named these variables better, it looks like there might be an issue, since it would seem that `generateConfigsForTableSpecs` would want `config` and not the currently generated config map.

Reply:
Yeah, I'll create a ticket to fix that separately. Most of this TableSpec/TableProvider code should go away post 1.0 anyway.